### ISSUE TITLE
[NavigationRailView] Enable NavigationRailView to expand like Flutter's NavigationRail Widget

### ISSUE THREAD

**Issue by Rishabh-More** [material-components/material-components-android#2469]
**Is your feature request related to a problem? Please describe.**
Our Android app primarily targets Android Tablet Devices. As such the app's UI is designed for Tablet Layouts. Inspired by Material Studies [Reply](https://material.io/design/material-studies/reply.html)'s `Expandable Navigation Rail` where the Navigation Rail can _expand to a dismissible Navigation Drawer_, we want to implement the same Navigation Method for our app.

**The Problem?**

Flutter's `NavigationRail` Widget has a boolean property named `extended` which expands the NavigationRail or collapses it. With the use of state management, you can control the opening or closing of your NavigationRail Widget:
```dart
NavigationRail(
  extended: true, // <-- Property to expand or collapse Navigation Rail
  selectedIndex: _selectedIndex,
  onDestinationSelected: (int index) {
    setState(() => _selectedIndex = index);
  },
  labelType: NavigationRailLabelType.none,
  destinations: const <NavigationRailDestination>[
    NavigationRailDestination(
      icon: Icon(Icons.favorite_border),
      selectedIcon: Icon(Icons.favorite),
      label: Text('First'),
    ),
    NavigationRailDestination(
      icon: Icon(Icons.bookmark_border),
      selectedIcon: Icon(Icons.book),
      label: Text('Second'),
    ),
    NavigationRailDestination(
      icon: Icon(Icons.star_border),
      selectedIcon: Icon(Icons.star),
      label: Text('Third'),
    ),
  ],
)
```

However, for **Android's** [NavigationRailView](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/navigationrail/NavigationRailView.java), there are **no methods** defined which can expand or collapse the NavigationRailView which I found very bizzare that Flutter has it, but Android's Material Components doesn't.

**Describe the solution you'd like**
It would be nice to have public methods which would expand & collapse the `NavigationRailView` to the default width of NavigationView, and maybe a property to set custom width for the expanded NavigationRail for fine control.

Or alternatively, you can kindly provide a custom NavigationRailView implementation that extends the material NavigationRailView with methods to expand or collapse, while we patiently wait for these methods to be added in a future release to android's NavigationRailView.

I'd definitely prefer having it added to the Component, and it will be surely worth the wait!!

**Comment by hunterstich** [comment]
Hi @Rishabh-More,

Thanks for the issue and interest! This is something we've thought about but design hasn't finalized as something our component officially supports. In the updated Reply mocks (https://m3.material.io/foundations/adaptive-design/overview) design has been leaning towards a NavigationRail paired with a drawer which slides in and over the rail for additional destinations instead of a rail that "extends". 

In other words, we don't have plans to support this at the time being. I'm going to close this issue but will re-open if design wants to explore this in the future.
